### ISSUE TITLE
[A11y BUG] nav drawer menu item traps tab incorrectly

### ISSUE THREAD

**Issue by seanwuapps** [getgoui/go-ui#104]
**Describe the bug**

Nav drawer menu opens and tab through, the focus stays on the first item and not changing

**To Reproduce**

1. go to https://go.components.style/docs/components/navigation/go-nav-drawer
2. in first demo, click any button to open nav drawer
3. press tab key a few times and observe focus highlight

**Expected behaviour**

The appropriate menu item should be focused after tab keypress

**Desktop (please complete the following information):**
 - OS: windows
 - Browser: chrome  98.0.4758.102
